broker/broker-security-impl/src/main/java/org/codice/ddf/broker/security/impl/SubjectInjectorPlugin.java

```diff
@@ -0,0 +1,121 @@
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.broker.security.impl;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.cache.Cache;
+import com.google.common.cache.CacheBuilder;
+import ddf.security.Subject;
+import ddf.security.assertion.SecurityAssertion;
+import ddf.security.service.SecurityManager;
+import ddf.security.service.SecurityServiceException;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import org.apache.activemq.artemis.api.core.Message;
+import org.apache.activemq.artemis.core.server.ServerSession;
+import org.apache.activemq.artemis.core.transaction.Transaction;
+import org.apache.activemq.artemis.protocol.amqp.broker.AMQPMessage;
+import org.apache.qpid.proton.amqp.messaging.ApplicationProperties;
+import org.codice.ddf.broker.security.api.BrokerMessageInterceptor;
+import org.codice.ddf.security.handler.api.UPAuthenticationToken;
+import org.codice.ddf.security.util.SAMLUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.w3c.dom.Element;
+
+public class SubjectInjectorPlugin implements BrokerMessageInterceptor {
+  private static final Logger LOGGER = LoggerFactory.getLogger(SubjectInjectorPlugin.class);
+
+  private static final Cache<String, Subject> SUBJECT_CACHE =
+      CacheBuilder.newBuilder().expireAfterWrite(1, TimeUnit.MINUTES).build();
+
+  private static SecurityManager securityManager;
+
+  private static Set<String> configuredAddresses;
+
+  @Override
+  public void handleMessage(
+      ServerSession session,
+      Transaction tx,
+      Message message,
+      boolean direct,
+      boolean noAutoCreateQueue) {
+    if (!configuredAddresses.contains(message.getAddress())) {
+      return;
+    }
+    String subjectAsString = getStringSubjectFromSession(session);
+
+    message.putStringProperty("subject", subjectAsString);
+    if (message instanceof AMQPMessage) {
+      Map applicationPropertiesMap =
+          ((AMQPMessage) message).getProtonMessage().getApplicationProperties().getValue();
+      applicationPropertiesMap.put("subject", subjectAsString);
+      ((AMQPMessage) message)
+          .getProtonMessage()
+          .setApplicationProperties(new ApplicationProperties(applicationPropertiesMap));
+    }
+  }
+
+  public void setConfiguredAddresses(Set<String> addresses) {
+    configuredAddresses = new HashSet<>(addresses);
+  }
+
+  private Element getSubjectAsElement(ServerSession session) {
+    try {
+      return SUBJECT_CACHE
+          .get(session.getUsername(), () -> this.cacheAndReturnSubject(session))
+          .getPrincipals()
+          .oneByType(SecurityAssertion.class)
+          .getSecurityToken()
+          .getToken();
+    } catch (ExecutionException e) {
+      LOGGER.warn("Could not get Subject from token", e);
```

Comment:
:pencil2: I think you could log `e.getCause()` instead since an `ExecutionException` just wraps the real exception that occurred.

Also, if your value loader function throws an unchecked exception (and I can't tell if it does) then `get()` will throw an `UncheckedExecutionException`, which you may or may not want to handle differently.